### ISSUE TITLE
Company admin able to see other company course participants and their reports

### ISSUE THREAD

**Issue by Parthajeet13** [iomad/iomad#2045]
Hello,

We are facing one challenge with the Multitenancy feature.

Currently, the courses are shared between different companies. When a Company admin from a company goes to a course in the participant, reports and grades tab, the company admin can see other company reports (these are the default Moodle reports). 

P.S.- When a company admin (or any user) is created he/she is added to a group (named like company) for all the courses. But the issue comes when the courses are shared. The company admin is able to see all the groups in the above-mentioned report and hence the list of users is visible.

The IOMAD version is 3.9 and we tried with the latest version but we didn't find any solution. Is there any way we can restrict the Moodle default reports w.r.t to the logged-in user's company?

Looking forward to your response.

Thanks,
Parthajeet~

**Comment by turf212** [comment]
Shared courses need to have forced separate groups on for the groups to be hidden.  Also - if you have changed any of the roles which are given to them (in the case of shared courses it will be Client course access == same as non-editing teacher core Moodle role) so that they can see outside of the group(s) they are assigned to, then they will be able to see outside of their group(s) and therefor all of the other company groups and users.

In other words - check you course settings for groups and the roles/permissions you have given the company managers in the course.
